File: demo1.m
addpath('data')
addpath('code')

%use Dimensionality.R to reduce the dimensionality

X = csvread('sim.csv',1,1);                %data after dimensionality reduction
root = 1;
k = 35;
epi = 0.45;
label = mylabelling(X, 1, k,root,epi);      %calculate labels
[Y1,B1,I1,obj1] = align(X,label, 1, 180);   %calculate pseudotime
Y1 = Y1/(max(Y1)-min(Y1));
Y1 = Y1+abs(min(Y1));
scatter(-X(:,1),X(:,2),25,Y1),hold on

File: demo1.R
# use Dimensionality.R to reduce the dimensionality

data <- R.matlab::readMat('sim1.mat')
# data after dimensionality reduction
X <- data$sim.DiffusionMap
X <- data.frame(dim1=X[,1],dim2=X[,2])

# calculate labels
result<-mylabelling_fix(X)
label<-result[["label"]]
cutoff<-result[["cutoff"]]
root<-result[["root"]]
plot(X,col=label)

# calculate pseudotime
pseudotime = align(X,label,root,c(40,100,cutoff))
ggplot(X, aes(x = dim1, y = dim2, color = pseudotime)) + geom_point() + scale_color_viridis_c(option = "viridis")
